### PR TITLE
feat: update to the latest pinning service api spec

### PR DESCRIPTION
**Pull request by aschmahmann**
> :point_right:  This needs https://github.com/ipfs/go-pinning-service-http-client/pull/4  to be merged first

Updates the spec to v1.0.0.

Codegen issues/hacking:
- Had to manually change the 4XX and 5XX code generation because it couldn't figure it out
- Wrapped `GetRequestid` as `GetRequestId` as it seemed more Go idiomatic, but could remove it

@petar @lidel 

~Note: Do not merge until we have confirmation this works against https://github.com/ipfs-shipyard/rb-pinning-service-api~
- ~Get { "error" : "unauthorized" } when trying to run an ls using my credentials~
   - ~This seems to indicate broken auth~
   - ~Also, this doesn't use the error types~

~Could be bugs on my end but want to confirm @andrew~

Resolved issues with the server.

### CODE REVIEW

**Review by lidel — CHANGES_REQUESTED** [review]
LGTM, only small concerns around ergonomics:

- main `./README.md` is missing steps on how to re-generate client 
- nit: [`./openapi/README.md`](https://github.com/ipfs/go-pinning-service-http-client/blob/feat/update-spec/openapi/README.md) starts with long Markdown from `description` fiels in spec YAML, which buries the lead: the client [overview](https://github.com/ipfs/go-pinning-service-http-client/blob/feat/update-spec/openapi/README.md#overview) docs start after 2/3 of the page - is there a generator flag to not include it? 
- nit: `Error` with only `ErrorError` looks weird, ideally we would flatten it and just have `Error`, if generator provides such feature. If not, we will have custom error responses in go-ipfs'   `/api/v0/pin/remote` anyway, so I think its fine.

**Review by lidel — CHANGES_REQUESTED** [review]
@aschmahmann please try re-generating with [v0.1.1](https://github.com/ipfs/pinning-services-api-spec/releases/tag/v0.1.1), it should resolve the problem with `ErrorError`.

**Review by lidel — CHANGES_REQUESTED** [review]
@aschmahmann / @petar this needs a fix for panics when remote endpoint is down (either network or DNS):

DNS lookup failure:

```console
$ ipfs pin remote service add test_invalid_url_dns_svc https://invalid-service.example.com fake_api_key
$ ipfs pin remote ls --service=test_invalid_url_dns_svc
panic: unexpected error type: Get "https://invalid-service.example.com/pins?limit=10": dial tcp: lookup invalid-service.example.com on 127.0.0.1:53: no such host

goroutine 66 [running]:
github.com/ipfs/go-pinning-service-http-client.httperr(0x0, 0x209ee40, 0xc0007fc570, 0x0, 0x0)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:373 +0x745
github.com/ipfs/go-pinning-service-http-client.(*Client).lsInternal(0xc0007ce018, 0x20d1880, 0xc0006db400, 0xc00065e180, 0x0, 0x0, 0x0, 0xc0000775d0, 0xb2ef1c, 0xc0000a8000)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:232 +0x338
github.com/ipfs/go-pinning-service-http-client.(*Client).Ls.func1(0xc00065e120, 0xc00065e0c0, 0xc0007ce018, 0x20d1880, 0xc0006db400, 0xc00065e180)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:160 +0x306
created by github.com/ipfs/go-pinning-service-http-client.(*Client).Ls
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:155 +0x1af

```


Service down:

```console

$ ipfs pin remote service add test_invalid_url_down_svc https://127.0.0.1:43534 fake_api_key
$ ipfs pin remote ls --service=test_invalid_url_down_svc
panic: unexpected error type: Get "https://127.0.0.1:43534/pins?limit=10": dial tcp 127.0.0.1:43534: connect: connection refused

goroutine 25 [running]:
github.com/ipfs/go-pinning-service-http-client.httperr(0x0, 0x209ee40, 0xc0000fe3c0, 0x0, 0x0)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:373 +0x745
github.com/ipfs/go-pinning-service-http-client.(*Client).lsInternal(0xc0000c8090, 0x20d1880, 0xc000890a80, 0xc0000e81e0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:232 +0x338
github.com/ipfs/go-pinning-service-http-client.(*Client).Ls.func1(0xc0000e8180, 0xc0000e8120, 0xc0000c8090, 0x20d1880, 0xc000890a80, 0xc0000e81e0)
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:160 +0x306
created by github.com/ipfs/go-pinning-service-http-client.(*Client).Ls
	pkg/mod/github.com/ipfs/go-pinning-service-http-client@v0.0.0-20201030162340-38f3baeb8878/client.go:155 +0x1af

```

File: client.go [hunk 1]
```diff
@@ -374,8 +369,18 @@ func getCIDEncoder() multibase.Encoder {
...
+		return e
 	}
+
+	ferr, ok := oerr.Model().(openapi.Failure)
+	if ok {
+		errJson, err := ferr.Error.MarshalJSON()
+		if err != nil {
+			return fmt.Errorf("failed to marshal Failure %w. Error was: {reason : %s, details : %s}", err, ferr.Error.GetReason(), ferr.Error.GetDetails())
+		}
+		return fmt.Errorf("%s", string(errJson))
```
Comment (aschmahmann) [inline client.go:381]: Should I also add in the status code (e.g. 401) or just have the reason + details?

File: client.go [hunk 2]
```diff
@@ -374,8 +369,18 @@ func getCIDEncoder() multibase.Encoder {
 func httperr(resp *http.Response, e error) error {
 	oerr, ok := e.(openapi.GenericOpenAPIError)
```
[the remark this answers is not in the document]
Comment (lidel) [reply client.go:370]: I think its good for now.
We can clean up errors in later RCs, for RC1 removing panic should be enough :+1:

File: client.go (hunk 1 above)
Comment (lidel) [reply client.go:381]: As this is the http client, it makes sense to include HTTP status code to help with debugging, but not a blocker.

Comment (aschmahmann) [reply client.go]: @lidel I removed the JSON marshalling here and did it manually. Added the HTTP status code text (i.e. "401 Unauthorized"), but could do the integer code to if you think that's nicer.

Comment (lidel) [reply client.go]: I believe an integer will be easier to work with

Comment (aschmahmann) [reply client.go]: updated

**Pull request merged**